pkg/parquetquery/nilIter.go

```diff
@@ -0,0 +1,201 @@
+package parquetquery
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	pq "github.com/parquet-go/parquet-go"
+)
+
+// NilAttributeIterator copies all functions of the sync iterator with just the next() function being different
+type NilAttributeIterator struct {
```

Comment:
I'm wondering if it would be better to implement the behavior of `NilAttributeIterator` in `SyncIterator` such that it can be turned on/off via a `SyncIteratorOption`?